Data.Access.Layer/Repositories/Contracts/IGenericRepository.cs

```diff
@@ -1,21 +1,24 @@
-﻿using System.Linq.Expressions;
+﻿using Data.Access.Layer.Models.Contracts;
+using System.Linq.Expressions;
 
 namespace Data.Access.Layer.Repositories.Contracts
 {
-    public interface IGenericRepository<T> where T : class, new()
+    public interface IGenericRepository<TEntity> where TEntity : class, IUser
     {
-        Task<IEnumerable<T?>> GetAll();
+        IQueryable<TEntity> Query();
 
-        Task<T?> GetById(Guid Id);
+        Task<ICollection<TEntity>> GetAllAsync();
 
-        Task<T?> Create(T entity);
+        Task<TEntity?> GetByIdAsync(Guid Id);
 
-        Task<int?> Delete(T entity);
+        Task<TEntity?> CreateAsync(TEntity entity);
 
-        Task<T?> Update(T entity, Guid id);
+        void Delete(TEntity entity);
```

Comment:
Should this method "return" Task instead of void?

Reply:
Delete is synchronous.